### ISSUE TITLE
Modify gateway (ST) to be able to implement PutBucketVersioning/GetBucketVersioning

### ISSUE THREAD

**Issue by mniewrzal** [storj/storj#6523]
Currently there is no straightforward way to use libuplink and implement proper versions of PutBucketVersioning/GetBucketVersioning gateway method. We need somehow expose those method to be able to implement correct behavior for them.

Acceptance Criteria:
* gateway st needs to be modified to give easy access to implement PutBucketVersioning/GetBucketVersioning
* its not scope of this ticket to implement logic for those methods

**Comment by shaupt131** [comment]
@mniewrzal and @amwolff to discuss if this is easy enough to do to include in the first release of Object Versioning, of if we add it as an enhancement to do later.

**Comment by shaupt131** [comment]
@mniewrzal and @amwolff have we come to a decision on this yet?  What's the expected lift to do this now as opposed to later?

**Comment by mniewrzal** [comment]
During last versioning office hours we discussed with @amwolff that modifying our `minio` fork is what we can do. Turns out that changes were relatively small and its already done with https://github.com/storj/minio/commit/730e38e5b7abe92a7438f6c3ee0312e5daf55a34. Versioning will be in `gateway-st` and `edge` from the first version.

After merging  https://review.dev.storj.io/c/storj/gateway/+/11801 we can close this issue.
